visualizations/frontend/filters/v2/TDatePicker.vue

```diff
@@ -0,0 +1,275 @@
+<template>
+    <div class="relative flex t-date-picker">
+        <!-- Date Picker -->
+        <base-date-picker
+            read-only
+            show-clear-button
+            confirm
+            confirm-text="Apply"
+            v-model="date"
+            ref="datePicker"
+            :range="mode === 'range'"
+            :format="dateFormat"
+            :range-separator="seperator"
+            @confirm="handleConfirm"
+            @pick="setCustomPreset"
+            @open="fetchLayerData"
+        >
+            <div class="presets-dropown" slot="sidebar">
+                <button
+                    v-for="preset in presets[mode]"
+                    :key="preset.key"
+                    :style="{ color: preset.key === selectedPreset.key ? '#1284e7' : '#73879c' }"
+                    class="cursor-pointer"
+                    @click="handlePreset(preset)"
+                >
+                    {{ preset.label }}
+                </button>
+            </div>
+
+            <div slot="icon-clear"></div>
+            <div slot="icon-calendar">
+                <div class="flex items-center gap-1">
+                    <button @click="handleClear">
+                        <close-icon />
```

Comment:
I'm not sure if anyone has already picked this up, but clicking the 'x' icon opens the calendar, and clicking the calendar icon does nothing. 


https://user-images.githubusercontent.com/1307818/186391489-e5211533-8773-4d7d-8450-84c195b7cf5e.mov

Reply:
> I'm not sure if anyone has already picked this up, but clicking the 'x' icon opens the calendar, and clicking the calendar icon does nothing. The pointer is a bit off in the recording, but I'm clicking the 'x' at first, and clicking the 'calendar' icon after.
> 
>  calendar-close-reload.mov

Strange but had to separate the loading icon for this to work, weird but works.